src/view/components/common/login-as/login-as.ts

```diff
@@ -54,4 +56,44 @@ export class LoginAsComponent extends BaseComponent {
         this.needPassword = this.loginService.getLoginAsUser($event.value).requestPassword || false;
     }
 
+    /**
+     * Filter the users displayed in the dropdown by comparing if
+     * the user name characters set on the drowpdown search box matches
+     * some on the user names set on the userlist variable loaded on the
+     * ngOnInit method
+     *
+     * @param event - The event with the query parameter to filter the users
+     */
+    filterUsers(event): void {
+        this.filteredLoginAsUsersResults = [];
+        for(let i = 0; i < this.userLists.length; i++) {
+            let user = this.userLists[i];
+            if(user.fullName.toLowerCase().indexOf(event.query.toLowerCase()) >= 0) {
+                this.filteredLoginAsUsersResults.push({
+                    label: user.fullName,
+                    value: user.userId,
+                });
+            }
+        }
+    }
+
+    /**
+     * Display all the existing login as users availables loaded on the
+     * userList variable initialized on the ngOnInit method
+     *
+     * @param event - The click event to display the dropdown options
+     */
+    handleLoginAsUsersDropdownClick(event): void {
+        this.filteredLoginAsUsersResults = [];
+        setTimeout(() => {
+            for(let i = 0; i < this.userLists.length; i++) {
+                let user = this.userLists[i];
+                this.filteredLoginAsUsersResults.push({
+                    label: user.fullName,
+                    value: user.userId,
+                });
+            }
+        }, 100);
+    }
+
```

Comment:
why we need the timeout?, we should look for another way to do this with the timeout